src/mobile/ios/Argon2/Argon2IOS.swift

```diff
@@ -18,18 +18,28 @@ class Argon2IOS: NSObject {
   ///   - password: Password to hash
   ///   - resolve: A JS Promise resolve block
   ///   - reject: A JS Promise reject block
-  @objc func hash(_ password: String, salt: String, params: [String: Any], resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
+  @objc func hash(_ password: NSArray, salt: String, params: [String: Any], resolver resolve: RCTPromiseResolveBlock, rejecter reject: RCTPromiseRejectBlock) {
     // Validate parameters
+
     if params["t_cost"] is Int && params["m_cost"] is Int && params["parallelism"] is Int && params["hashLength"] is Int {
       // Resolve the hash
-      let h = Argon2Core.argon2Hash(password: password, salt: salt, params: params)
-      resolve(h)
+      let passwordLength = password.count as Int
+      var passwordInt8 = [Int8](repeating: 0, count: passwordLength)
+      for i in 0...passwordLength - 1 {
+        passwordInt8[i] = password[i] as! Int8
```

Comment:
Xcode gives a warning here: `Cast from 'UInt8' to unrelated type 'Int8' always fails`

Try replacing lines 27-30 with:

```swift
let passwordInt8 = password.map { Int8(bitPattern: $0) }
```

This will be a bit cleaner than a `for-in` loop and uses `init(bitPattern:)` instead of a force downcast (which basically tells the compiler "I know what I'm doing, let me try this anyway and if it doesn't work, trigger a crash")